src/iter.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019 multiset developers
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+#![warn(missing_docs)]
+
+use std::borrow::Borrow;
+use std::marker::PhantomData;
+
+/// An iterator over the items of a `MultiSet`.
+///
+/// This `struct` is created by the [`iter`](super::HashMultiSet::iter) method on
+/// [`HashMultiSet`](super::HashMultiSet) or [`BTreeMultiSet`](super::BTreeMultiSet).
+pub struct Iter<K: Clone, V: Borrow<usize>, InnerIter: Iterator<Item = (K, V)>> {
+    pub(crate) iter: InnerIter,
+    pub(crate) duplicate: Option<<InnerIter as Iterator>::Item>,
+    pub(crate) duplicate_index: usize,
+    pub(crate) duplicate_back: Option<<InnerIter as Iterator>::Item>,
+    pub(crate) duplicate_index_back: usize,
+    pub(crate) len: usize,
+    pub(crate) _ghost: PhantomData<*const (K, V)>,
+}
+
+impl<K: Clone, V: Borrow<usize>, InnerIter: Iterator<Item = (K, V)> + ExactSizeIterator>
+    Iter<K, V, InnerIter>
+{
+    pub(crate) fn new(iter: InnerIter, len: usize) -> Self {
+        Iter {
+            iter,
+            duplicate: None,
+            duplicate_index: 0,
+            duplicate_back: None,
+            duplicate_index_back: 0,
+            len,
+            _ghost: PhantomData,
+        }
+    }
+}
+
+impl<K: Clone, V: Borrow<usize>, InnerIter: Iterator<Item = (K, V)> + Clone> Clone
+    for Iter<K, V, InnerIter>
+where
+    <InnerIter as Iterator>::Item: Clone,
+{
+    fn clone(&self) -> Iter<K, V, InnerIter> {
```

Comment:
Why would `#[derive(Clone)]` not work?

Reply:
Oh, I don't know :(
Most likely it came from the previous implementation.
Thanks